scrapscript.py

```diff
@@ -2482,6 +2482,64 @@ def flat_command(args: argparse.Namespace) -> None:
     sys.stdout.buffer.write(serializer.output)
 
 
+def server_command(args: argparse.Namespace) -> None:
+    import http.server
+    import socketserver
+    import hashlib
+    import os.path
+
+    dir = os.path.abspath(args.directory)
+    if not os.path.isdir(dir):
+        print(f"Error: {dir} is not a valid directory")
+        sys.exit(1)
+
+    scraps = {}
+    for root, _, files in os.walk(dir):
+        for file in files:
+            if file.endswith(".scrap"):
+                file_path = os.path.join(root, file)
+                rel_path = os.path.relpath(file_path, dir)
+                rel_path_without_ext = os.path.splitext(rel_path)[0]
+                with open(file_path, "r") as f:
+                    try:
+                        program = parse(tokenize(f.read()))
+                        serializer = Serializer()
+                        serializer.serialize(program)
+                        serialized = bytes(serializer.output)
+                        scraps[rel_path_without_ext] = serialized
+                        logger.debug(f"Loaded {rel_path_without_ext}")
+                        file_hash = hashlib.md5(serialized).hexdigest()
+                        scraps[f"${file_hash}"] = serialized
```

Comment:
I don't think we should put them in the same namespace unless you enforce that filenames cannot start with `$`